x/merkledb/db.go

```diff
@@ -233,53 +226,65 @@ func newDatabase(
 	}
 
 	// mark that the db has not yet been cleanly closed
-	err = trieDB.metadataDB.Put(cleanShutdownKey, didNotHaveCleanShutdown)
+	err = trieDB.baseDB.Put(cleanShutdownKey, didNotHaveCleanShutdown)
 	return trieDB, err
 }
 
 // Deletes every intermediate node and rebuilds them by re-adding every key/value.
 // TODO: make this more efficient by only clearing out the stale portions of the trie.
 func (db *merkleDB) rebuild(ctx context.Context) error {
 	db.root = newNode(nil, RootPath)
-	if err := db.nodeDB.Delete(rootKey); err != nil {
-		return err
-	}
```

Comment:
Was this previously a bug? Seems like `rebuild` isn't handling the case that the root node had a value right?

Reply:
I modified the `Test_MerkleDB_DB_Rebuild` test and this was previously a bug. Can we add a regression test for this?